exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillMetaImpl.java

```diff
@@ -78,19 +111,237 @@ private MetaResultSet s(String s) {
+        String tableCat,
+        String tableSchem,
+        String tableName,
+        String columnName,
+        int dataType,
+        String typeName,
+        Integer columnSize,
+        Integer decimalDigits,
+        Integer numPrecRadix,
+        int nullable,
+        Integer charOctetLength,
+        int ordinalPosition,
+        String isNullable) {
+      this.tableCat = tableCat;
+      this.tableSchem = tableSchem;
+      this.tableName = tableName;
+      this.columnName = columnName;
+      this.dataType = dataType;
+      this.typeName = typeName;
+      this.columnSize = columnSize;
+      this.decimalDigits = decimalDigits;
+      this.numPrecRadix = numPrecRadix;
+      this.nullable = nullable;
+      this.charOctetLength = charOctetLength;
+      this.ordinalPosition = ordinalPosition;
+      this.isNullable = isNullable;
+    }
+
+    @Override
+    public String getName() {
+      return columnName;
+    }
+  }
+
+  private static LikeFilter newLikeFilter(final Pat pattern) {
+    if (pattern == null || pattern.s == null) {
+      return null;
+    }
+
+    return LikeFilter.newBuilder().setPattern(pattern.s).build();
+  }
+
+  /**
+   * Quote the provided string as a LIKE pattern
+   *
+   * @param v the value to quote
+   * @return a LIKE pattern matching exactly v, or {@code null} if v is {@code null}
+   */
+  private static Pat quote(String v) {
+    if (v == null) {
+      return null;
+    }
+
+    StringBuilder sb = new StringBuilder(v.length());
+    for(int index = 0; index<v.length(); index++) {
+      char c = v.charAt(index);
+      switch(c) {
+      case '%':
+      case '_':
+      case '\\':
+        sb.append('\\').append(c);
+        break;
+
+      default:
+        sb.append(c);
+      }
+    }
+
+    return Pat.of(sb.toString());
+  }
+
+  // Overriding fieldMetaData as Calcite version create ColumnMetaData with invalid offset
+  protected static ColumnMetaData.StructType drillFieldMetaData(Class<?> clazz) {
+    final List<ColumnMetaData> list = new ArrayList<>();
+    for (Field field : clazz.getFields()) {
+      if (Modifier.isPublic(field.getModifiers())
+          && !Modifier.isStatic(field.getModifiers())) {
+        NotNull notNull = field.getAnnotation(NotNull.class);
+        boolean notNullable = (notNull != null || field.getType().isPrimitive());
+        list.add(
+            drillColumnMetaData(
+                AvaticaUtils.camelToUpper(field.getName()),
+                list.size(), field.getType(), notNullable));
+      }
+    }
+    return ColumnMetaData.struct(list);
+  }
+
+
+  protected static ColumnMetaData drillColumnMetaData(String name, int index,
+      Class<?> type, boolean notNullable) {
+    TypeInfo pair = TypeInfo.get(type);
+    ColumnMetaData.Rep rep =
+        ColumnMetaData.Rep.VALUE_MAP.get(type);
+    ColumnMetaData.AvaticaType scalarType =
+        ColumnMetaData.scalar(pair.sqlType, pair.sqlTypeName, rep);
+    return new ColumnMetaData(
```

Comment:
yes, it is fine. This method is only used for ResultSetMetaData, it just creates column metadata based on a pojo field.and none of the types used here have precision nor scale. 

Metadata for queries are handled in a different way and are already derived from the BatchSchema.